app/code/Magento/CompareListGraphQl/Model/Resolver/AddItemsToCompareList.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Resolver;
+
+use Magento\Catalog\Model\CompareList as ModelCompareList;
```

Comment:
I would recommend decoupling this resolver into smaller chunks. We may make the resolver 'thin' by moving all the business logic to a separate class(es). In that way we will reduce the number of dependencies in this resolver